dev/io.openliberty.org.apache.myfaces.4.0_fat/test-applications/AnnotationLiteralsTest.war/resources/annotationLiterals.xhtml

```diff
@@ -0,0 +1,132 @@
+<!DOCTYPE html>
+<!--
+    Copyright (c) 2022 IBM Corporation and others.
+    All rights reserved. This program and the accompanying materials
+    are made available under the terms of the Eclipse Public License 2.0
+    which accompanies this distribution, and is available at
+    http://www.eclipse.org/legal/epl-2.0/
+    
+    SPDX-License-Identifier: EPL-2.0
+
+    Contributors:
+        IBM Corporation - initial API and implementation
+ -->
+<html xmlns="http://www.w3.org/1999/xhtml" xmlns:f="jakarta.faces.core"
+	xmlns:h="jakarta.faces.html" xmlns:c="jakarta.tags.core"
+	xmlns:cb="http://io.openliberty/test/annotation/literal/clientBehavior">
+
+<h:head>
+	<title>Value outputs from Annotation Literals</title>
+</h:head>
+
+<h:body>
+	<c:if test="#{param.testApplicationMap}">
+		<h:outputText value="#{testBean.applicationMap}" />
+	</c:if>
+
+	<c:if test="#{param.testFacesBehavior}">
+		<h:outputText value="#{testBean.facesBehavior}" />
+	</c:if>
+
+	<c:if test="#{param.testFacesConverter}">
+		<h:outputText value="#{testBean.facesConverter}" />
+	</c:if>
+
+	<c:if test="#{param.testFacesConfig}">
+		<h:outputText value="#{testBean.facesConfig}" />
+	</c:if>
+
+	<c:if test="#{param.testFacesDataModel}">
+		<h:outputText value="#{testBean.facesDataModel}" />
+	</c:if>
+
+	<c:if test="#{param.testFacesValidator}">
+		<h:outputText value="#{testBean.facesValidator}" />
+	</c:if>
+
+	<c:if test="#{param.testHeaderMap}">
+		<h:outputText value="#{testBean.headerMap}" />
+	</c:if>
+
+	<c:if test="#{param.testHeaderValuesMap}">
+		<h:outputText value="#{testBean.headerValuesMap}" />
+	</c:if>
+
+	<c:if test="#{param.testInitParameterMap}">
+		<h:outputText value="#{testBean.initParameterMap}" />
+	</c:if>
+
+	<c:if test="#{param.testManagedProperty}">
+		<h:outputText value="#{testBean.managedProperty}" />
+	</c:if>
+
+	<c:if test="#{param.testRequestCookieMap}">
+		<h:outputText value="#{testBean.requestCookieMap}" />
+	</c:if>
+
+	<c:if test="#{param.testRequestMap}">
+		<h:outputText value="#{testBean.requestMap}" />
+	</c:if>
+
+	<c:if test="#{param.testRequestParameterMap}">
+		<h:outputText value="#{testBean.requestParameterMap}" />
+	</c:if>
+
+	<c:if test="#{param.testRequestParameterValuesMap}">
+		<h:outputText value="#{testBean.requestParameterValuesMap}" />
+	</c:if>
+
+	<c:if test="#{param.testSessionMap}">
+		<h:outputText value="#{testBean.sessionMap}" />
+	</c:if>
+
+	<c:if test="#{param.testView}">
+		<h:outputText value="#{testBean.view}" />
+	</c:if>
+
+	<c:if test="#{param.testWebsocketEvent}">
+		<h:form id="open">
+			<h:commandButton id="websocket" value="websocket" onClick="faces.push.open('testChannel')" >
```

Comment:
Websocket isn't working due to `onClick` --  it should be lowercase.

Reply:
Hmm.  My local FAT run ran into a problem...? 

`2023-02-01-15:23:56:738 Exception invoking jsConstructor `
Caused by: java.lang.NoSuchMethodError: 'org.eclipse.jetty.client.HttpClient org.eclipse.jetty.websocket.client.WebSocketClient.getHttpClient()'

Reply:
Faces messages aren't showing when I test manually. I need to investigate it more .

Reply:
Took too long to figure out 

1) As I mentioned in another comment, these CDI TypeLiterals selects don't work.

2) The requests to the EventBean aren't hitting because they hidden before the c:if #{param.testWebsocketEvent)

You'll need to the query back to the server to ensure the if is evaluated . as so: 
```
			<h:commandButton id="submit" value="submit" action="#{eventBean.logOpenWebsocket()}" >
				<f:param name = "testWebsocketEvent" value = "#{param.testWebsocketEvent}" /> 
				<f:ajax/>
			</h:commandButton>
```

3)  I forgot that c:if is a JSTL tag. It may not always play nice with JSF.   (They are executed at different times -- build time vs render time) That's another topic entirely.